### DISCUSSION TITLE
How are you automating the testing your PDFs? Cypress? Percy?

### DISCUSSION THREAD

**Discussion by acomito** [wojtekmaj/react-pdf#1499] [Q&A]
We use react-pdf and have maybe 120 different variations of the PDF that we need to test.

I've been thinking through ways we can effectively test this. The two that have come to mind so far are:

1. A mix of cypress and percy-- where we use snapshots

2. comparing a known hard coded base64 version of the PDF with the base64 version of the PDF results of a test (i.e. use cypress to go e2e and then generate/download the pdf and compare it to what we know the outcome should be in base64)

**Comment by wojtekmaj** [comment]
It looks like you confused React-PDF with [@react-pdf/renderer](https://github.com/diegomura/react-pdf). Please start a discussion in their repo instead.

On a personal note, comparing base64 seems like a bad idea - even a smallest bugfix would result in everything red. I'd go with Playwright and a mix of text-based and screenshot-based assertions.
